spec/app/ics-027-interchain-accounts/README.md

```diff
@@ -501,34 +498,29 @@ function onChanCloseConfirm(
 
 ```typescript
 function OnRecvPacket(packet Packet) {
-    	ack := NewResultAcknowledgement([]byte{byte(1)})
+    	ack = NewResultAcknowledgement([]byte{byte(1)})
 
 	// only attempt the application logic if the packet data
 	// was successfully decoded
-	if ack.Success() {
-		switch data.Type {
-	          case types.EXECUTE_TX:
-		        msgs, err := types.DeserializeTx(data.Data)
-		        if err != nil {
-			      return err
-		        }
-
-                        // ExecuteTx calls the AuthenticateTx function defined above 
-		        if err = ExecuteTx(ctx, packet.SourcePort, packet.DestinationPort, packet.DestinationChannel, msgs); err != nil {
-			      return err
-		        }
-
-		        return nil
-	          default:
-		        return ErrUnknownDataType
-	        }
-	        if err != nil {
-                    ack = NewErrorAcknowledgement(err.Error())
-                }
-	}
-
-	// NOTE: acknowledgment will be written synchronously during IBC handler execution.
-	return ack
+  switch data.Type {
+  case types.EXECUTE_TX:
+  msgs, err := types.DeserializeTx(data.Data)
```

Comment:
Isn't the norm "throwing exceptions" in this ics spec "pseudo code".

It seems modelled heavily on TypeScript

Reply:
We don't want to throw an exception in this case and abort the transaction, we want to write an error acknowledgement and commit the state changes

Reply:
Understood. But the idiom `msg, err := fn()` is very go. The rest of the ibc specs are modelled on TypeScript.

I would say:

```ts
try {
  const msg = deserializeTx(data.data)
  // ... other stuff
} catch (err) {
  // return error packet
}
```

would look more in line with the other specs (but still needs some adjustment).

I believe the specs were intentionally not written in Go to keep a clear distinction between them and the implementation. Whether that was a good decision I will leave to you and @cwgoes to discuss. I just like consistency in the style.